### ISSUE TITLE
twisted.web.test.test_xml.BrokenHTMLTests.test_brokenAttributeName relies on stable dict ordering

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8016]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8016|
|Type|enhancement|
|Created|2015-09-06 05:44:31Z|
|Branch|https://github.com/twisted/twisted/tree/xml-stable-8016|

For this test to pass on PyPy/non-stable-dict platforms, it should not rely on CPython's default dict ordering.

<details><summary>Searchable metadata</summary>

```
trac-id__8016 8016
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__PyPy_support PyPy-support
branch__branches_xml_stable_8016 branches/xml-stable-8016
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1441518271317496 1441518271317496
changetime__1441595550634008 1441595550634008
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [45643]) Branching to xml-stable-8016.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

PyPy builder spun (https://buildbot.twistedmatrix.com/builders/ubuntu15.04-pypy25/builds/38) , the rest of the builders spun too.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Looks good. Thanks! Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45648]) Merge xml-stable-8016: Make the microdom tests not rely on stable dict ordering

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8016](https://github.com/twisted/twisted/issues/8016)
